website/docs/docs/build/ref-metrics-in-filters.md

```diff
@@ -0,0 +1,163 @@
+    from 
+        {{ ref('fct_model_runs') }}
+    group by 
+        account_id
+),
+
+activated_accounts as (
+    select
+        count(distinct account_id) as activated_accounts
+    from 
+        {{ ref('dim_accounts') }}
+    left join 
+        data_models_per_user 
+    on 
+        {{ ref('dim_accounts') }}.account_id = data_models_per_user.account
+    where 
+        data_models_per_user.data_model_runs > 5
+)
+
+select
+    *
+from 
+    activated_accounts
+```
+
+This previous SQL query calculates the number of `activated_accounts` by using the `data_model_runs` metric as a dimension for the user entity. It filters based on the metric value scoped to the account entity. You can express this logic natively in the MetricFlow specification.
+
+### Reference a metric in a filter
+
+You can recreate this pattern in MetricFlow by referencing a metric in the `where` filter for another metric using the `Metric()` object syntax. The function for referencing a metric takes a metric name and exactly one entity:
+
+```yaml
+{{ Metric('metric_name', group_by=['Entity']) }}
+```
+
+Using the same `activated_accounts` metric as an example:
+- Create two semantic models, `model_runs` and `accounts`. 
+- Then create a `measure` and `metric` to count data model runs, and another measure to count users.
+- Specify the foreign entity `account` in the `model_runs` semantic model.
+
+```yaml
+semantic_models:
+  - name: model_runs
+    .... # Placeholder for other configurations
+	entities:
+      - name: model_run
+        type: primary
+      - name: account
+        type: foreign
+    measures:
+      - name: data_model_runs
+        agg: sum
+        expr: 1
+        create_metric: true
+
+  - name: accounts
+  	.... # Placeholder for other configurations
+	entities:
+      - name: account
+        type: primary
+    measures:
+      - name: accounts
+        agg: sum
+        expr: 1
+```
+
+- Then create the `Activated Accounts` metric by filtering accounts that have more than five data model runs:
+
+```yaml
+metrics:
+ - name: activated_accounts
+	 label: Activated Accounts
+   type: simple
+   type_params:
+	   measure: accounts
+	 filter: |
+		 {{ Metric('data_model_runs', group_by=['account']) }} > 5
+```
+
+Let’s break down the SQL that will be generated from this metric definition when I run the following query from the CLI `dbt sl query --metrics activated_accounts` 
+
+First, the filter `{{ Metric('data_models', group_by=['account']) }}` will generate SQL similar to the `data_models_per_user` sub query shown above.
+
+```sql
+select
+  sum(1) as data_model_runs
+  , account
+ from data_model_runs
+ group by 
+	 account
+```
+
+We will then join this query to the query generated by the `accounts` measure on the group by elements and apply the filter conditions.
+
+```sql
+select
+	sum(1) as activated_accounts
+from accounts
+left join (
+select
+    sum(1) as data_model_runs
+  , account
+ from data_model_runs
```

Comment:
should this be `{{ ref('data_model_runs') }}`?

Reply:
Hmm not sure it makes sense here since the the SQL generated by MetricFlow will reference a table directly, not a `ref()`.